### ISSUE TITLE
Split proving / verification parameter loading into different subsystems.

### ISSUE THREAD

**Issue by nathan-at-least** [zcash/zcash#599]
Proving key:

* takes a long time to load.
* only needed for the wallet.
* only needed in the wallet when spending confidential coins (aka "protected buckets").

Verification key:

* takes ~1s to load.
* is required immediately upon node start up as soon as blocks come in *either* from the network or during the startup process (which revalidates the past ~1000 blocks).  (in fact, the public alpha branch crashed in the block-revalidation step due to validation parameter being initialized too late.)

Meanwhile, the existing component architecture is:

* block validation must occur immediately upon startup, where by contrast:
* the wallet is optional, and:
* the wallet loads in a background thread, and there's already an RPC protocol for signaling that the wallet in a node is not ready.

**Comment by ebfull** [comment]
Honestly, I think we should do this for 1.0. @defuse has already done most of the work for it anyway. This will make our rpc tests run somewhat faster.

**Comment by daira** [comment]
+1

**Comment by nathan-at-least** [comment]
How much effort is it to move loading the proving key into the wallet startup, so that it does not load if wallet is turned off, and it does not block chain validation when turned on?

We strongly prefer that to lazy loading, and we moved away from the original lazy loading.

Reasons to avoid lazy loading:

* it's hard to understand operational cost by reading code. (In the old lazy-load version of libzerocash tests, some tests would do throw-away operations to trigger load as a side-effect, and other code did not do this and incorrectly counted loading as part of the cost of prooving for cost measurement.)
* if proof-sequencing is non-deterministic or hard to figure out, then so is parameter loads. (eg: if unit tests are run in parallel, which test will load the keys?)
* it adds a UX burden in a naive wallet (the first user's pour takes extra long, whereas if they start their wallet, the proving params should be loaded eagerly, assuming they want to do a poor)


(The old lazy-loaded test code would do dummy-operations in some cases to trigger a load, which is confusing code. Other code didn't do this pre-emptive load, and incorrectly measured load time as part of a proof cost, etc...)

**Comment by nathan-at-least** [comment]
BTW- I would accept PR #783 even though I dislike that behavior, simply because it's ready and dramatically improves our automated test times, so if the eager-wallet-component load is too much work, I'll ACK #783.
